scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -51,24 +52,49 @@ class ExperimentalScanner(
     val storageWriters: List<ScanStorageWriter>,
     val packageProvenanceResolver: PackageProvenanceResolver,
     val nestedProvenanceResolver: NestedProvenanceResolver,
-    val scannerWrappers: List<ScannerWrapper>
+    val scannerWrappers: Map<DataEntity, List<ScannerWrapper>>
 ) {
     init {
-        require(scannerWrappers.isNotEmpty()) {
+        require(scannerWrappers.isNotEmpty() && scannerWrappers.any { it.value.isNotEmpty() }) {
             "At least one ScannerWrapper must be provided."
         }
     }
 
     suspend fun scan(ortResult: OrtResult): OrtResult {
         val startTime = Instant.now()
 
-        val result = scan(
-            (ortResult.getProjects().map { it.toPackage() } + ortResult.getPackages().map { it.pkg }).toSet()
-        )
+        val projectScannerWrappers = scannerWrappers[DataEntity.PROJECTS].orEmpty()
```

Comment:
Have you considered following the Open-Closed principle here, so that we can later-on add more sophisticated methods for assigning a wrapper to a project or package without changing anything in the code of this class?

What I have in mind is injecting an instance of:
```
interface ScannerWrapperResolver {
       fun resolveWrapper(pkg: CuratedPackage, isProject: Boolean): List<ScannerWrapper>
}
```

What do you think?

Reply:
Is there any real use case for this or is this more a theoretic question? Because I currently strive for feature parity with the default scanner implementation.

Reply:
I can imagine real use cases, but the design implications this would bring would be already worth it IMO even if there was no real use case. It becomes more modular and better testable and a bit less repetetive and slightly more generic. IIUC correctly it'd not add complexity, but maybe I missed something.

Reply:
I agree that it would be nice to have this, but it's too complex for me to implement right now, given that we don't have a use case at hand. Especially since it's unclear based on what the wrappers should be selected, should it be based on static configuration or custom code like you suggested on Slack? And what information should be available for the selection (just `pkg` and `isProject` as suggested above is certainly not enough)?
To me the priority is to cover the features we currently have to finally reach a state where we can replace the current scanner implementation.

Reply:
Let's keep it for now then.

Reply:
BTW, this topic is basically documented as https://github.com/oss-review-toolkit/ort/issues/2819, and we actually do have a concrete use-case at hand by now as well: We'd like to use a snippet-scanner (as opposed to a regular license scanner) if a package has the `isModified` property set.